backend/remote-state/http/client.go

```diff
@@ -0,0 +1,298 @@
+package http
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	multierror "github.com/hashicorp/go-multierror"
+
+	"github.com/hashicorp/terraform/state"
+	"github.com/hashicorp/terraform/state/remote"
+)
+
+// RemoteClient is used by "state/remote".State to read and write
+// blobs representing state.
+// Implements "state/remote".ClientLocker
+type RemoteClient struct {
+	client *http.Client
+
+	// The fields below are set from configure
+	address       string
+	updateMethod  string
+	lockAddress   string
+	unlockMethod  string
+	lockMethod    string
+	unlockAddress string
+	username      string
+	password      string
+
+	lockID       string
+	jsonLockInfo []byte
+}
+
+// Get blah
+func (c *RemoteClient) Get() (*remote.Payload, error) {
+	// Convert address to type URL
+	addressURL, _ := url.Parse(c.address)
+
+	resp, err := c.httpRequest("GET", addressURL, nil, "get state")
```

Comment:
Use `http.MethodGet` constant

Reply:
They are using it this way. Didn't wanted to change it... I guess I can change it to http.MethodGet since there is no clear view of LOCK and UNLOCK. 
https://github.com/hashicorp/terraform/blob/master/state/remote/http.go#L227